python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -3161,6 +3163,103 @@ def add_backfill(self, partition_backfill: "PartitionBackfill") -> None:
     def update_backfill(self, partition_backfill: "PartitionBackfill") -> None:
         self._run_storage.update_backfill(partition_backfill)
 
+    @public
+    def launch_backfill(
+        self,
+        asset_graph: "BaseAssetGraph",
+        asset_selection: Optional[Sequence[AssetKey]] = None,
+        partitions_by_assets: Optional[
+            Sequence[Tuple[AssetKey, Optional[Sequence[Tuple[str, str]]]]]
+        ] = None,
+        partition_names: Optional[Sequence[str]] = None,
+        tags: Mapping[str, str] = {},
+        all_partitions: bool = False,
+        title: Optional[str] = None,
+        description: Optional[str] = None,
+    ) -> str:
+        """Launch a set of partition backfill runs.
+
+        Either partition_names must not be None or all_partitions must be True but not both.
+
+        Args:
+            asset_graph (BaseAssetGraph): The asset graph for the backfill.
+            asset_selection (Optional[Sequence[AssetKey]]): List of asset keys to backfill.
+            partitions_by_assets (Optional[Sequence[Tuple[AssetKey, Optional[Sequence[Tuple[str, str]]]]]]):
+                List of partitions for each asset key to backfill.
+            partition_names (Optional[Sequence[str]]): List of partition names to backfill. Only
+                valid if `asset_selection` is provided.
+
+        Returns:
+            str: The ID of the backfill.
+        """
+        # TODO(deepyaman): Abstract logic shared with `dagster-graphql`.
```

Comment:
```suggestion
```

Actually, as I got to implementing more of this, maybe they're sufficiently different that this isn't worth it.